test/test_numpy.py

```diff
@@ -219,3 +220,72 @@ def test_align():
         for j in range(3):
             for k in range(4):
                 assert x(i=i, j=j, k=k) == y(i=i, j=j, k=k)
+
+
+def unary_eval(symbol, x):
+    if symbol in ['~', '-']:
+        return eval('{} x'.format(symbol))
+    if isinstance(x, np.ndarray):
+        return getattr(funsor.ops, symbol)(x)
+    return getattr(x, symbol)()
+
+
+@pytest.mark.parametrize('dims', [(), ('a',), ('a', 'b')])
+@pytest.mark.parametrize('symbol', [
+    '~', '-', 'abs', 'sqrt', 'exp', 'log', 'log1p', 'sigmoid',
+])
+def test_unary(symbol, dims):
+    sizes = {'a': 3, 'b': 4}
+    shape = tuple(sizes[d] for d in dims)
+    inputs = OrderedDict((d, bint(sizes[d])) for d in dims)
+    dtype = 'real'
+    data = np.array(np.random.rand(*shape)) + 0.5
+    if symbol == '~':
+        data = data.astype(bool)
+        dtype = 2
+    expected_data = unary_eval(symbol, data)
```

Comment:
You could special case here
```py
if symbol == 'sigmoid':
    expected_data = scipy.special.expit(data)
else:
    expected_data = unary_eval(symbol, data)
```

Reply:
In `unary_eval`, I added `if isinstance(x, np.ndarray): return getattr(funsor.ops, symbol)(x)` so the issue does not happen at this line. :(